### PR TITLE
Speed up code and avoid unnecessary MatchData objects

### PR DESCRIPTION
**Pull request by AaronLasseigne**
This runs faster and creates fewer objects.

A more detailed discussion including some benchmarks can be found here: https://github.com/schneems/rails/commit/1bf50badd943e684a56a03392ef0ddafefca0ad7#commitcomment-12572839

cc: @schneems

### CODE REVIEW

**Comment by schneems** [comment]
Thanks! You rock!

**Comment by AaronLasseigne** [comment]
Woot! Thanks for the merge.

File: activesupport/lib/active_support/inflector/inflections.rb [hunk 1]
```diff
@@ -49,7 +49,7 @@ def add(words)
         end
 
         def uncountable?(str)
-          @regex_array.detect {|regex| regex.match(str) }
+          @regex_array.any? { |regex| str =~ regex }
```
Comment (sgrif) [inline activesupport/lib/active_support/inflector/inflections.rb]: Should this use `Regex#===`?

Comment (rafaelfranca) [inline activesupport/lib/active_support/inflector/inflections.rb]: @schneems already did later.

Comment (sgrif) [inline activesupport/lib/active_support/inflector/inflections.rb]: I suppose it's 0 allocations regardless.

Comment (sgrif) [inline activesupport/lib/active_support/inflector/inflections.rb]: Gotcha. Thanks.

Comment (schneems) [inline activesupport/lib/active_support/inflector/inflections.rb]: It's debatable which is better, here's another thread on the topic: https://github.com/JuanitoFatas/fast-ruby/pull/62 I think either is fine. Probably didn't need the patch I sent in.

**Pull request merged**